### PR TITLE
[BadgeDrawable] add horizontal and vertical offset to control position

### LINKED ISSUE #630: [BadgeDrawable] add margin for Badge
I'm using badge with `BottomNavigationView` ,  
but the badge position only change  by  `gravity` ,
so I can't  control the position of badge 

Is there any plan to add  something like `margin`  to control the position of badge?

### PR DESCRIPTION
**Pull request by Chaoyang0201**
add horizontal and vertical offset to control badge  position
close #630

### CODE REVIEW

**Comment by ldjcmu** [comment]
Hi Chaoyang0201,

Thanks for the PR. We're working with our design team to verify that this is how they want to support BadgeDrawable position configurations (and how that overlaps with gravity). We'll update with any needs/suggestions we get from them.

**Comment by Chaoyang0201** [comment]
> Could you please add a unit test?

Sure, I will update later

File: lib/java/com/google/android/material/badge/BadgeDrawable.java [hunk 1]
```diff
@@ -720,16 +734,16 @@ private void calculateCenterAndBounds(
...
         break;
       case BOTTOM_END:
       case TOP_END:
       default:
         badgeCenterX =
             ViewCompat.getLayoutDirection(anchorView) == View.LAYOUT_DIRECTION_LTR
-                ? anchorRect.right + halfBadgeWidth - inset
-                : anchorRect.left - halfBadgeWidth + inset;
+                ? anchorRect.right + halfBadgeWidth - inset - savedState.badgeHorizontalOffset
+                : anchorRect.left - halfBadgeWidth + inset + savedState.badgeHorizontalOffset;
```
Comment (wcshi) [inline lib/java/com/google/android/material/badge/BadgeDrawable.java]: typo

**Comment by Chaoyang0201** [comment]
> Thank you for the PR, I am in the process of pulling in your change.

OK,  tell me if anything else needs

**Pull request closed without merging**